modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/downsampling/DeleteSourceAndAddDownsampleToDS.java

```diff
@@ -101,31 +119,20 @@ ClusterState execute(ClusterState state) {
                         + "] but it is instead part of data stream ["
                         + sourceParentDataStream.getName()
                         + "]";
-                if (sourceParentDataStream.getWriteIndex().getName().equals(sourceBackingIndex)) {
-                    String errorMessage = String.format(
-                        Locale.ROOT,
-                        "index [%s] is the write index for data stream [%s] and cannot be replaced",
-                        sourceBackingIndex,
-                        sourceParentDataStream.getName()
-                    );
-                    throw new IllegalStateException(errorMessage);
-                }
-                if (sourceIndexMeta != null) {
-                    // both indices exist, let's copy the origination date from the source index to the downsample index
-                    Metadata.Builder newMetaData = Metadata.builder(state.getMetadata());
-                    TimeValue generationLifecycleDate = dataStream.getGenerationLifecycleDate(sourceIndexMeta);
-                    assert generationLifecycleDate != null : "write index must never be downsampled, or replaced";
-                    IndexMetadata updatedDownsampleMetadata = copyDataStreamLifecycleState(
-                        sourceIndexMeta,
-                        downsampleIndexMeta,
-                        generationLifecycleDate.millis()
-                    );
+                // both indices exist, let's copy the origination date from the source index to the downsample index
+                Metadata.Builder newMetaData = Metadata.builder(state.getMetadata());
+                TimeValue generationLifecycleDate = dataStream.getGenerationLifecycleDate(sourceIndexMeta);
+                assert generationLifecycleDate != null : "write index must never be downsampled, or replaced";
```

Comment:
nit: This is the error message for another assert, not this one.

Reply:
Not it's not. But if it's confusing I'm happy to change it? 

The generation lifecycle data is null only for write indices. If we get a `null` generation date inside the task that's meant to replace a source index with a donwsample index it means ... somehow there's an attempt to downsample and then replace the write index of a data stream.

Reply:
Yeah it would be good to clarify for future maintainers -- it's confusing since you've already thrown an IllegalStateException above if the source index is the write index.

Reply:
Sure. Done.